libctru/include/3ds/services/frd.h

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file frd.h
+ * @brief Friend Services
+ */
+ #pragma once
+#include <3ds.h>
+
+#define FRIENDS_SCREEN_NAME_SIZE 0x16   // 11 (0x16 because UTF-16)
+#define FRIENDS_COMMENT_SIZE 0x22       // 16 (0x21 because UTF-16 + null character)
+#define FRIEND_LIST_SIZE 0x64           // 100 (Number of Friends)
+#define FRIEND_MII_STORE_DATA_SIZE 0x60 // 96 (Mii data)
+
+/// Friend key data
+typedef struct
+{
+   u32 principalId;
+   u32 padding;
+   u64 localFriendCode;
+} FriendKey;
+
+/// Structure containing basic Mii information.
+typedef struct 
+{
+    u32 mii_id;
+    u64 system_id;
+    u32 cdate;
+    u8 mac[0x6];
+    u16 padding;
+    u16 misc1;
+    u16 mii_name[0xB];
+    u8 width;
+    u8 height;
+    u32 misc2;
+    u32 unknown1;
+    u32 misc3;
+    u32 unknown2;
+    u8 allow_copy;
+    u8 unknown3[0x7];
+    u16 author[0xB];
+} MiiData;
+
+/// Friend profile data
+typedef struct
+{
+    u8 region;      // The region code for the hardware.
```

Comment:
You should probably change this to (CFG_Region)[https://github.com/smealum/ctrulib/blob/master/libctru/include/3ds/services/cfgu.h#L17]

Reply:
Enums inside struct types that are going to be used for interfacing with external code is not exactly a good idea; because enum sizes are not defined by the standard and it is up to the compiler and ABI to decide which size they have by default. For instance, ARM (32-bit) defines they shall have the smallest size that will fit all values in the enum, but ARM (64-bit) defines them to have at least the size of int.

Reply:
Oh okay